reporter/otlp_reporter.go

```diff
@@ -241,18 +230,28 @@ func (r *OTLPReporter) FrameMetadata(args *FrameMetadataArgs) {
 		defer frameMapLock.WUnlock(&frameMap)
 
 		sourceFile := args.SourceFile
-		if sourceFile == "" {
-			// The new SourceFile may be empty, and we don't want to overwrite
-			// an existing filePath with it.
+		sourceLine := args.SourceLine
+		functionOffset := args.FunctionOffset
+		if sourceFile == "" || sourceLine == 0 || functionOffset == 0 {
+			// Some of the new metadata fields may be unset, and we don't want to overwrite
+			// existing data with it.
 			if s, exists := (*frameMap)[addressOrLine]; exists {
-				sourceFile = s.filePath
+				if sourceFile == "" {
+					sourceFile = s.filePath
+				}
+				if sourceLine == 0 {
+					sourceLine = s.lineNumber
+				}
+				if functionOffset == 0 {
+					functionOffset = s.functionOffset
+				}
```

Comment:
The reporter doesn't know about internals and whether metadata possibly becomes "updated". If you say that the interpreter code only reports all data fields or doesn't report at all, we can drop this "update" code completely. The (current) kernel symbols are known to only report the function name. Since FileIDs of kernel and interpreters are different, we should be fine.

Reply:
Reverted the changes.